provider/rosa_operator_roles_data_source.go

```diff
@@ -0,0 +1,217 @@
+
+	// Get the collection of clusters:
+	clustersClient := parent.connection.ClustersMgmt().V1().Clusters()
+	awsInquiries := parent.connection.ClustersMgmt().V1().AWSInquiries()
+
+	// Create the resource:
+	result = &RosaOperatorRolesDataSource{
+		logger:         parent.logger,
+		clustersClient: clustersClient,
+		awsInquiries:   awsInquiries,
+	}
+	return
+}
+
+func (t *RosaOperatorRolesDataSource) Read(ctx context.Context, request tfsdk.ReadDataSourceRequest,
+	response *tfsdk.ReadDataSourceResponse) {
+	// Get the state:
+	state := &RosaOperatorRolesState{}
+	diags := request.Config.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	stsOperatorRolesList, err := t.awsInquiries.STSCredentialRequests().List().Send()
+	if err != nil {
+		t.logger.Error(ctx, "Failed to get operator list")
+		return
+	}
+	stsOperatorMap := make(map[string]*cmv1.STSOperator)
+	stsOperatorRolesList.Items().Each(func(stsCredentialRequest *cmv1.STSCredentialRequest) bool {
+		t.logger.Debug(ctx, "Operator name: %s, namespace %s, service account %s",
+			stsCredentialRequest.Operator().Name(),
+			stsCredentialRequest.Operator().Namespace(),
+			stsCredentialRequest.Operator().ServiceAccounts(),
+		)
+		// The key can't be stsCredentialRequest.Operator().Name() because of constants between
+		// the name of `ingress_operator_cloud_credentials` and `cloud_network_config_controller_cloud_credentials`
+		// both of them includes the same Name `cloud-credentials` and it cannot be fixed
+		stsOperatorMap[stsCredentialRequest.Operator().Namespace()] = stsCredentialRequest.Operator()
+		return true
+	})
+	get, err := t.clustersClient.Cluster(state.ClusterID.Value).Get().SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Can't find cluster",
+			fmt.Sprintf(
+				"Can't find cluster with identifier '%s': %v",
+				state.ClusterID.Value, err,
+			),
+		)
+		return
+	}
+	object := get.Body()
+	sts, ok := object.AWS().GetSTS()
+	if ok {
+		for _, operatorRole := range sts.OperatorIAMRoles() {
+			r := OperatorIAMRole{
+				Name: types.String{
+					Value: operatorRole.Name(),
+				},
+				Namespace: types.String{
+					Value: operatorRole.Namespace(),
+				},
+				RoleARN: types.String{
+					Value: operatorRole.RoleARN(),
+				},
+				RoleName: types.String{
+					Value: getRoleName(state.OperatorRolePrefix.Value, operatorRole.Namespace(), operatorRole.Name()),
+				},
+				PolicyName: types.String{
+					Value: getRoleName(state.AccountRolePrefix.Value, operatorRole.Namespace(), operatorRole.Name()),
+				},
+				ServiceAccounts: getServiceAccount(stsOperatorMap[operatorRole.Namespace()].ServiceAccounts(), operatorRole.Namespace()),
+			}
+			state.OperatorIAMRoles = append(state.OperatorIAMRoles, &r)
+		}
+	}
+	// Save the state:
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+}
+
+func getRoleName(prefix string, namespace string, name string) string {
```

Comment:
Could you add some TODO or some form of tag so we can identify later what could be setup in a separate repo?

Reply:
Sure great idea!